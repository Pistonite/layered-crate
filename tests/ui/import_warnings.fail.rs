use layered_crate::layers;

#[layers]
mod src {
    #[depends_on(y)]
    #[depends_on(x)]
    extern crate z; // this should have warning since z is never used

    extern crate y;
    extern crate x;
}

fn main() {
    compile_error!("make it fail so we can check the warnings")
}
