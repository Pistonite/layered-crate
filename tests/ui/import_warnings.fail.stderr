error: make it fail so we can check the warnings
  --> tests/ui/import_warnings.fail.rs:14:5
   |
14 |     compile_error!("make it fail so we can check the warnings")
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused import: `z`
 --> tests/ui/import_warnings.fail.rs:7:18
  |
7 |     extern crate z; // this should have warning since z is never used
  |                  ^
  |
  = note: `#[warn(unused_imports)]` on by default

warning: unused import: `x`
 --> tests/ui/import_warnings.fail.rs:6:18
  |
6 |     #[depends_on(x)]
  |                  ^
